docs/api.md

```diff
@@ -0,0 +1,3 @@
+# API reference
+
+::: linmod
```

Comment:
I wasn't able to see any code docs until I listed out the modules like this:

```suggestion
::: linmod.models

::: linmod.eval

::: linmod.visualize
```

Do we want all of these on one page, or perhaps separate pages?